src/features/user/Settings/EditProfile.tsx

```diff
@@ -212,42 +211,20 @@ export default function EditProfile({}: Props) {
     }
     if (contextLoaded && token) {
       try {
-        putAuthenticatedRequest(
+        const res = await putAuthenticatedRequest(
           `/app/profile`,
           bodyToSend,
           token,
-          impersonatedEmail,
-          handleError
-        )
-          .then((res) => {
-            console.log(res);
-            if (res.code !== 400) {
-              setSeverity('success');
-              setSnackbarMessage(ready ? t('editProfile:profileSaved') : '');
-              handleSnackbarOpen();
-              setIsUploadingData(false);
-              setUser(res);
-            } else {
-              setSeverity('error');
-              setSnackbarMessage(
-                ready ? t('editProfile:profileSaveFailed') : ''
-              );
-              handleSnackbarOpen();
-              setIsUploadingData(false);
-            }
-          })
-          .catch((error) => {
-            setSeverity('error');
-            setSnackbarMessage(ready ? t('editProfile:profileSaveFailed') : '');
-            handleSnackbarOpen();
-            setIsUploadingData(false);
-            console.log(error);
-          });
-      } catch (e) {
-        setSeverity('error');
-        setSnackbarMessage(ready ? t('editProfile:profileSaveFailed') : '');
+          impersonatedEmail
+        );
+        setSeverity('success');
+        setSnackbarMessage(ready ? t('editProfile:profileSaved') : '');
         handleSnackbarOpen();
         setIsUploadingData(false);
+        setUser(res);
+      } catch (err) {
+        setIsUploadingData(false);
+        setErrors(handleError(err as APIError));
```

Comment:
General error messages are getting triggered because API does not return a custom error.

The system is designed in such a way that if custom error is present then it will get the first preference or else a general error based on status code will be prompt to the user.

Jorgo is working on sending custom errors.